modular_skyrat/modules/modular_implants/code/nifsofts/hivemind.dm

```diff
@@ -0,0 +1,263 @@
+
+		to_chat(hivemind_owner, span_abductor("[linked_mob] has left your Hivemind"))
+		to_chat(linked_mob, span_abductor("You have left [hivemind_owner]'s Hivemind"))
+
+	qdel(user_network)
+
+	//add in a function that scans the mind_links in the network list and remove the user from each one of them.
+	. = ..()
+
+/datum/nifsoft/hivemind/activate()
+	. = ..()
+	if(!active)
+		qdel(linked_keyboard)
+		linked_keyboard = null
+		return TRUE
+
+	linked_keyboard = new
+	linked_keyboard.connected_network = active_network
+	linked_mob.put_in_hands(linked_keyboard)
+	linked_keyboard.source_user = linked_mob
+
+	linked_mob.visible_message(span_notice("The [linked_keyboard] materializes in [linked_mob]'s hands"), span_notice("The [linked_keyboard] appears in your hands"))
+	return TRUE
+
+/datum/action/innate/hivemind_config
+	name = "Hivemind Configuration Settings"
+	button_icon = 'modular_skyrat/master_files/icons/mob/actions/action_backgrounds.dmi'
+	background_icon_state = "android"
+	icon_icon = 'modular_skyrat/master_files/icons/mob/actions/actions_nif.dmi'
+	button_icon_state = "phone"
+
+/datum/action/innate/hivemind_config/Activate()
+	. = ..()
+	var/datum/component/mind_linker/nif/link = target
+
+	var/choice = tgui_input_list(owner, "Chose your option", "Hivemind Configuration Menu", list("Link a user","Remove a user","Change Hivemind color","Change active Hivemind","Leave a Hivemind", "Toggle Invites"))
+	if(!choice)
+		return
+
+	switch(choice)
+		if("Link a user")
+			link.invite_user()
+
+		if("Remove a user")
+			link.remove_user()
+
+		if("Leave a Hivemind")
+			leave_hivemind()
+
+		if("Change active Hivemind")
+			change_hivemind()
+
+		if("Change Hivemind color")
+			link.change_chat_color()
+
+		if("Toggle Invites")
+			toggle_invites()
+
+/datum/action/innate/hivemind_config/proc/change_hivemind()
+	var/mob/living/carbon/human/user = owner
+	var/datum/nifsoft/hivemind/hivemind = user.find_nifsoft(/datum/nifsoft/hivemind)
+
+	var/datum/component/mind_linker/nif/new_active_hivemind = tgui_input_list(user, "Chose a Hivemind to set as active", "Switch Hivemind", hivemind.network_list)
+	if(!new_active_hivemind)
+		return FALSE
+
+	hivemind.active_network = new_active_hivemind
+	to_chat(user, span_abductor("You are now sending messages to [new_active_hivemind.name]"))
+
+/datum/action/innate/hivemind_config/proc/leave_hivemind()
+	var/mob/living/carbon/human/user = owner
+	var/datum/nifsoft/hivemind/hivemind = user.find_nifsoft(/datum/nifsoft/hivemind)
+
+	var/list/network_list = hivemind.network_list
+	network_list -= hivemind.user_network
+
+	var/datum/component/mind_linker/nif/hivemind_to_leave = tgui_input_list(user, "Chose a Hivemind to disconnect from", "Remove Hivemind", network_list)
+	if(!hivemind_to_leave)
+		return FALSE
+
+	to_chat(hivemind_to_leave.parent, span_abductor("[user] has been removed from your Hivemind"))
+	to_chat(user, span_abductor("You have left [hivemind_to_leave.parent]'s hivemind"))
```

Comment:
```suggestion
	to_chat(user, span_abductor("You have left [hivemind_to_leave.parent]'s hivemind."))
```
is hivemind proper or not? seen varying captialization

Reply:
It's proper, I think.